cli/cmd/deploy/deploy_test.go

```diff
@@ -0,0 +1,204 @@
+package deploy_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/google/go-github/v71/github"
+	"github.com/google/uuid"
+	"github.com/joho/godotenv"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/admin/database"
+	"github.com/rilldata/rill/admin/testadmin"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	"github.com/rilldata/rill/cli/testcli"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestManagedDeploy(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode")
+	}
+	err := godotenv.Load("../../../.env")
+	require.NoError(t, err)
+
+	adm := testadmin.New(t)
+
+	_, c := adm.NewUser(t)
+	u1 := testcli.New(t, adm, c.Token)
+
+	result := u1.Run(t, "org", "create", "github-test")
+	require.Equal(t, 0, result.ExitCode)
+
+	// deploy the project
+	tempDir := t.TempDir()
+	os.WriteFile(filepath.Join(tempDir, "rill.yaml"), []byte(`compiler: rillv1
+display_name: Untitled Rill Project
+olap_connector: duckdb`), 0644)
+
+	result = u1.Run(t, "project", "deploy", "--interactive=false", "--org=github-test", "--project=rill-mgd-deploy", "--skip-deploy=true", "--path="+tempDir)
+	require.Equal(t, 0, result.ExitCode, result.Output)
+
+	// verify the project is correctly created
+	resp, err := c.GetProject(t.Context(), &adminv1.GetProjectRequest{
+		OrganizationName: "github-test",
+		Name:             "rill-mgd-deploy",
+	})
+	require.NoError(t, err)
+	require.Equal(t, "rill-mgd-deploy", resp.Project.Name)
+
+	// get a github client
+	installationID, err := adm.Admin.Github.ManagedOrgInstallationID()
+	require.NoError(t, err)
+	ghClient := adm.Admin.Github.InstallationClient(installationID, nil)
+
+	// cleanup repo
+	t.Cleanup(func() {
+		owner, repo, ok := gitutil.SplitGithubRemote(resp.Project.GitRemote)
+		require.True(t, ok, "invalid github remote: %s", resp.Project.GitRemote)
+		_, err = ghClient.Repositories.Delete(context.Background(), owner, repo)
+		require.NoError(t, err, "failed to delete github repo %s/%s: %w", owner, repo, err)
+	})
+
+	// redeploy the same project with changes
+	err = os.Mkdir(filepath.Join(tempDir, "models"), 0755)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(tempDir, "models/model.sql"), []byte(`SELECT 1 AS one`), 0644)
+	require.NoError(t, err)
```

Comment:
nit: consider using a map of files and having a util function similar to `testruntime.PutFiles`: https://github.com/rilldata/rill/blob/68143db83dfb6aa9062fa8471a3af30087e262ab/runtime/testruntime/reconcile.go#L18